packages/log/src/log_helpers.ts

```diff
@@ -53,55 +59,115 @@ export function getFormatFromEnv(): LogFormat {
     }
 }
 
+type SanitizeDataOptions = {
+    maxDepth?: number;
+    gradualLimitFactor?: number;
+    maxStringLength?: number;
+    maxArrayLength?: number;
+    maxFields?: number;
+    preferredFieldsMap?: Record<PropertyKey, number>;
+    truncationSuffix?: string;
+    truncationFlagKey?: string;
+};
+
 /**
- * Limits given object to given depth and escapes function with [function] string.
+ * Sanitizes given object based on the given options.
  *
  * ie. Replaces object's content by '[object]' and array's content
- * by '[array]' when the value is nested more than given limit.
+ * by '[array]' when the value is nested more than given depth limit.
  */
-export function limitDepth<T>(record: T, depth: number, maxStringLength?: number): T | undefined {
+export function sanitizeData(data: unknown, options: SanitizeDataOptions): unknown {
```

Comment:
In the future, we might wanna use `node-inspect-extracted` too instead of our custom implementation